src/Frontend/Widgets/FilePane.vala

```diff
@@ -113,32 +116,44 @@ namespace Taxi {
 
         private Gee.List<string> get_marked_row_uris () {
             var uri_list = new Gee.ArrayList<string> ();
-            foreach (Gtk.Widget row in list_box.get_children ()) {
-                if (row.get_data<Gtk.CheckButton> ("checkbutton").get_active ()) {
-                    uri_list.add (current_uri.to_string () + "/" + row.get_data<string> ("name"));
+
+            Gtk.ListBoxRow row = null;
+            var row_index = 0;
+
+            do {
+                row = list_box.get_row_at_index (row_index);
+                if (((FileRow)row).active) {
```

Comment:
Here causes the following warning on runtime:

```
** (com.github.alecaddd.taxi:10910): CRITICAL **: 23:39:56.053: taxi_file_row_get_active: assertion 'self != NULL' failed
```

This is because we call `get_row_at_index` with off-by-one index after last row and there are null check here

Reply:
You should add the type check or null check and plus avoid do-while like this:

```vala
            unowned var row = list_box.get_first_child ();
            while (row != null) {
                if (row is FileRow) {
                    unowned var file_row = (FileRow) row;
                    if (file_row.active) {
                        uri_list.add (current_uri.to_string () + "/" + file_row.file_name);
                    }
                }

                row = row.get_next_sibling ();
            }
```